src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -105,111 +135,212 @@ class FlightTaskAutoFollowTarget : public FlightTask
+	 * @param sticks Sticks object to get RC commanded values for adjustments
+	 */
+	void update_rc_adjusted_follow_height(const Sticks &sticks);
+
+	/**
+	 * Update the Follow distance based on RC commands
+	 *
+	 * If the drone's distance error to the setpoint is within the an user adjustment error time window
+	 * follow_distance will be adjusted with a speed proportional to user RC command
+	 *
+	 * @param sticks Sticks object to get RC commanded values for adjustments
+	 * @param drone_to_target_vector Tracked follow distance variable reference which will be updated to the new value
+	 */
+	void update_rc_adjusted_follow_distance(const Sticks &sticks, const Vector2f &drone_to_target_vector);
+
+	/**
+	 * Update the Follow angle based on RC commands
+	 *
+	 * If the drone's orbit angle in relation to target is within the an user adjustment error time window
+	 * away from the orbit angle setpoint, follow_angle will be adjusted with a speed proportional to user RC command
+	 *
+	 * @param sticks Sticks object to get RC commanded values for adjustments
+	 * @param measured_angle Measured current drone's orbit angle around the target (depends on tracked target orientation for reference)
+	 * @param tracked_orbit_angle_setpoint Rate constrained orbit angle setpoint value from last command
+	 */
+	void update_rc_adjusted_follow_angle(const Sticks &sticks, const float measured_orbit_angle, const float tracked_orbit_angle_setpoint);
+
+	/**
+	 * Update the Second Order Target Position + Velocity Filter to track kinematically feasible target position and velocity
+	 *
+	 * @param follow_target_estimator Received value from alpha-beta-gamma target estimator filter output
+	 */
+	void update_target_position_velocity_filter(const follow_target_estimator_s &follow_target_estimator);
+
+	/**
+	 * Calculate the tracked target orientation and overwrite the tracked target orientation if necessary
+	 *
+	 * Note : Filtered target velocity is generated via 2nd order filter can have overshooting behaviors
+	 * when target stops it's motion. This can generate a target velocity output that is opposite direction to where target was heading originally.
+	 * To check if the filtered velocity is staying true to target's actual motion, unfiltered velocity needs to be taken into account. Since during
+	 * overshoot, the unfiltered velocity stays close to 0 (indicating target already stopped), therefore not triggering a target orientation setting.
+	 *
+	 * @param current_target_orientation [rad] Tracked target orientation value that will be over-written with new orientation
+	 * @param target_velocity [m/s, m/s] Filtered Target velocity from which we will calculate target orientation
+	 * @param target_velocity_unfiltered [m/s, m/s] Unfiltered Target velocity that aids in verifying if filtered velocity is accurate
+	 */
+	void update_target_orientation(float &current_target_orientation, const Vector2f &target_velocity, const Vector2f &target_velocity_unfiltered) const;
```

Comment:
this comment is still not addressed https://github.com/PX4/PX4-Autopilot/pull/19260#discussion_r862709971

1. more intuitive to *return* the target orientation (not do param[in,out])
2. still defining [m/s, m/s] unnecessarily: